test/asyncapiSchemaFormatParser_test.js

```diff
@@ -98,4 +109,38 @@ describe('asyncapiSchemaFormatParser', function() {
 
     expect(async () => await parser.parse(parsedInput)).to.not.throw();
   });
+
+  it('should deep clone schema into x-parser-original-payload', async function() {
```

Comment:
This test says it "should deep clone schema" but we're actually not checking if it's a deep clone, or am I missing something? If you remove the `deepClone` call above and just leave it like this:

```js
message['x-parser-original-payload'] = message.payload;
```

I think this test still passes.

Reply:
It verifies that it's a deepclone as the parser would have modified the original payload. So if it's not a deep clone, `message['x-parser-original-payload']` fails because the payload gets modified, and so does the 'original' because they just share a reference. If I remove the deep clone, the test fails because the extra parser pieces have now appeared.

```
 AssertionError: expected { Object (type, properties, ...) } to deeply equal { Object (type, properties) }
      + expected - actual

           "id": {
             "description": "Id of the streetlight."
             "minimum": 0
             "type": "integer"
      -      "x-parser-schema-id": "<anonymous-schema-2>"
           }
           "lumens": {
             "description": "Light intensity measured in lumens."
             "minimum": 0
             "type": "integer"
      -      "x-parser-schema-id": "<anonymous-schema-3>"
           }
           "sentAt": {
             "description": "Date and time when the message was sent."
             "format": "date-time"
             "type": "string"
      -      "x-parser-schema-id": "<anonymous-schema-4>"
           }
         }
         "type": "object"
      -  "x-parser-schema-id": "<anonymous-schema-1>"
       }
```

Reply:
I see. Thanks for explaining.

What about adding an extra test saying that whatever is on `x-parser-original-payload` !== `message.payload`? If we compare an object with itself using `to.deep.equal`, it's going to pass but it's not a clone.